src/shogun/machine/StochasticGBMachine.cpp

```diff
@@ -187,31 +189,25 @@ bool CStochasticGBMachine::train_machine(CFeatures* data)
 
 	for (auto i : SG_PROGRESS(range(m_num_iter)))
 	{
-		// apply subset
-		if (m_subset_frac!=1.0)
-			apply_subset(feats,interf);
+		const auto result = get_subset(feats, interf);
```

Comment:
why all these changes? shouldnt we focus on the view first?

Reply:
this is taken from michele's pr, we need small refactor on this to deploy views

Reply:
ok,
I technically would prefer first adding the view and then have a second Pr to deploy it.
But I think others see that differently, so ok leave it in :)